lib/driver.js

```diff
@@ -1207,19 +1207,25 @@ class XCUITestDriver extends BaseDriver {
           return {device, realDevice: false, udid: device.udid};
         }
       } else {
-        // make sure it is a connected device. If not, the udid passed in is invalid
-        const devices = await getConnectedDevices();
-        this.log.debug(`Available devices: ${devices.join(', ')}`);
-        if (!devices.includes(this.opts.udid)) {
-          // check for a particular simulator
-          this.log.debug(`No real device with udid '${this.opts.udid}'. Looking for a simulator`);
-          try {
-            const device = await getSimulator(this.opts.udid, {
-              devicesSetPath: this.opts.simulatorDevicesSetPath,
-            });
-            return {device, realDevice: false, udid: this.opts.udid};
-          } catch (ign) {
-            throw new Error(`Unknown device or simulator UDID: '${this.opts.udid}'`);
+        // If the session specified this.opts.webDriverAgentUrl with a real device,
+        // we can assume the user prepared the device properly already.
+        if (this.isRealDevice() && this.opts.webDriverAgentUrl) {
```

Comment:
Possibly this real device check is not necessary

Reply:
yes